test/630_proxy_dns_test.sh

```diff
@@ -6,46 +6,39 @@ C1=10.2.0.78
 C2=10.2.0.34
 C1_NAME=c1.weave.local
 C2_NAME=seetwo.weave.local
-
-boot_containers() {
-  proxy_start_container_with_dns $HOST1 -e WEAVE_CIDR=$C2/24 -dt --name=c2 -h $C2_NAME
-  proxy_start_container_with_dns $HOST1 -e WEAVE_CIDR=$C1/24 -dt --name=c1
-}
-
-kill_containers() {
+STATIC=static.name
+STATIC_IP=10.9.9.9
+
+do_assert_resolution() {
+  proxy_start_container_with_dns $HOST1 -e WEAVE_CIDR=$C2/24 -dt --name=c2 --add-host $STATIC:$STATIC_IP -h $C2_NAME
+  proxy_start_container_with_dns $HOST1 -e WEAVE_CIDR=$C1/24 -dt --name=c1 --add-host $STATIC:$STATIC_IP
+  $1 $HOST1 c1 $C2_NAME $C2
+  assert_dns_record $HOST1 c1 $STATIC $STATIC_IP
```

Comment:
not just though; the test also attempts to resolve `$C2_NAME`, which will happen via DNS.